File: 01-collatz/collatz.lua
function collatz(n)
    local cnt = 0
    while n > 1 do
        n = (n%2>0) and 3*n+1 or n/2
        cnt = cnt+1
    end
    return cnt
end

n = os.getenv('MAXN')
sum = 0
for i=1,n do
    sum = sum + collatz(i)
end
print(string.format("sum=%d, avg=%f", sum, sum/n))

File: 01-collatz/collatz.java
class Collatz {
	static int collatz(int x) {
		int cnt = 0;
		while (x > 1) {
			x = x % 2 != 0 ? 3 * x + 1 : x / 2;
			cnt++;
		}
		return cnt;
	}
	
	public static void main(String[] argv) {
		int maxN = Integer.parseInt(System.getenv("MAXN"));
		int sum = 0;
		for (int i = 0; i <= maxN; i++) {
		    sum += collatz(i);
		}
		System.out.printf("sum=%d, avg=%.2f%n", sum, sum * 1.0 / maxN);
	}
}
